pkg/cluster/clustertest/cluster.go

```diff
@@ -0,0 +1,264 @@
+	}
+	return nil
+}
+
+// Delete implements interface function Delete
+func (mc *MockedCluster) Delete(key string) error {
+	if mc.MockedDelete != nil {
+		return mc.MockedDelete(key)
+	}
+	return nil
+}
+
+// DeletePrefix implements interface function DeletePrefix
+func (mc *MockedCluster) DeletePrefix(prefix string) error {
+	if mc.MockedDeletePrefix != nil {
+		return mc.MockedDeletePrefix(prefix)
+	}
+	return nil
+}
+
+// STM implements interface function STM
+func (mc *MockedCluster) STM(apply func(concurrency.STM) error) error {
+	if mc.MockedSTM != nil {
+		return mc.MockedSTM(apply)
+	}
+	return nil
+}
+
+// Watcher implements interface function Watcher
+func (mc *MockedCluster) Watcher() (cluster.Watcher, error) {
+	if mc.MockedWatcher != nil {
+		return mc.MockedWatcher()
+	}
+	return nil, nil
+}
+
+// Syncer implements interface function Syncer
+func (mc *MockedCluster) Syncer(pullInterval time.Duration) (*cluster.Syncer, error) {
+	if mc.MockedSyncer != nil {
+		return mc.MockedSyncer(pullInterval)
+	}
+	return nil, nil
+}
+
+// Mutex implements interface function Mutex
+func (mc *MockedCluster) Mutex(name string) (cluster.Mutex, error) {
+	if mc.MockedMutex != nil {
+		return mc.MockedMutex(name)
+	}
+	return nil, nil
+}
+
+// CloseServer implements interface function CloseServer
+func (mc *MockedCluster) CloseServer(wg *sync.WaitGroup) {
+	if mc.MockedCloseServer != nil {
+		mc.MockedCloseServer(wg)
+	}
+}
+
+// StartServer implements interface function StartServer
+func (mc *MockedCluster) StartServer() (chan struct{}, chan struct{}, error) {
+	if mc.MockedStartServer != nil {
+		return mc.MockedStartServer()
+	}
+	return nil, nil, nil
+}
+
+// Close implements interface function Close
+func (mc *MockedCluster) Close(wg *sync.WaitGroup) {
+	if mc.MockedClose != nil {
+		mc.MockedClose(wg)
+	}
+}
+
+// PurgeMember implements interface function PurgeMember
+func (mc *MockedCluster) PurgeMember(member string) error {
+	if mc.MockedPurgeMember != nil {
+		return mc.MockedPurgeMember(member)
+	}
+	return nil
+}
+
+// MockedSTM is a mocked cocurrency.STM
+type MockedSTM struct {
+	// embed concurrency.STM for commit & reset
+	concurrency.STM
+	MockedGet func(key ...string) string
+	MockedPut func(key, val string, opts ...clientv3.OpOption)
+	MockedRev func(key string) int64
+	MockedDel func(key string)
+}
+
+// Get implements STM.Get
+func (stm *MockedSTM) Get(key ...string) string {
+	if stm.MockedGet != nil {
+		return stm.MockedGet(key...)
+	}
+	return ""
+}
+
+// Put implements STM.Put
+func (stm *MockedSTM) Put(key, val string, opts ...clientv3.OpOption) {
+	if stm.MockedPut != nil {
+		stm.MockedPut(key, val, opts...)
+	}
+}
+
+// Rev implements STM.Rev
+func (stm *MockedSTM) Rev(key string) int64 {
+	if stm.MockedRev != nil {
+		return stm.MockedRev(key)
+	}
+	return 0
+}
+
+// Del implements STM.Del
+func (stm *MockedSTM) Del(key string) {
+	if stm.MockedRev != nil {
+		stm.MockedDel(key)
+	}
+}
```

Comment:
Just noting here that after this PR this `MockedCluster` could be used in `pkg/filter/headerlookup/headerlookup_test.go` and some other Filter tests too to make the tests run faster.

Reply:
https://github.com/megaease/easegress/pull/590